spec/bookshelf-manager-v1.yaml

```diff
@@ -0,0 +1,186 @@
+openapi: 3.0.3
+info:
+  title: Bookshelf Manager API
+  version: '1.0.0'
+  description: |-
+    A simple backend Bookshelf Manager API to manage books and items on a bookshelf.
+tags:
+  - name: books
+    description: Access and manage book information
+  - name: admin
+    description: Operations that require a user with admin privileges
+paths:
+  /books:
+    get:
+      tags:
+        - books
+      summary: Find all books
+      description: Returns a list of all existing books
+      responses:
+        '200':
+          description: Successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Book'
+        '404':
+          description: No books found
+    post:
+      tags:
+        - books
+      summary: Add a new book
+      description: Add a new book
+      security:
+        - ApiKeyHeader: []
+        - ApiKeyQuery: []
+      requestBody:
+        description: Create a new book
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Book'
+        required: true
+      responses:
+        '201':
+          description: Successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Book'
+        '400':
+          description: Invalid book ID supplied
+    delete:
+      tags:
+        - admin
+      summary: Delete all books
+      description: Delete all books
+      security:
+        - OAuth2: [admin]
+          ApiKeyHeader: []
+        - OAuth2: [admin]
+          ApiKeyQuery: []
+      responses:
+        '204':
+          description: Successful operation
+        '401':
+          description: Not authenticated
+        '403':
+          description: Access token does not have the required scope
+  /books/{bookId}:
+    get:
+      tags:
+        - books
+      summary: Find book by ID
+      description: Returns a single book
+      parameters:
+        - name: bookId
+          in: path
+          description: ID of book to return
+          required: true
+          schema:
+            type: integer
+            format: int64
+      responses:
+        '200':
+          description: Successful retrieval of the book
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Book'
+        '400':
+          description: Invalid book ID supplied
+        '404':
+          description: Book not found
+    put:
+      tags:
+        - books
+      summary: Update an existing book by ID
+      description: Update an existing book by ID
+      security:
+        - ApiKeyHeader: []
+        - ApiKeyQuery: []
```

Comment:
For POST and PUT endpoints, we should require API key for tracking and logging purposes. 

And the dashes indicate different ways to authenticate with API key, see the OpenAPI doc: https://swagger.io/docs/specification/authentication/#multiple